sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala

```diff
@@ -70,7 +70,8 @@ case class TransformWithStateExec(
       // TODO: check if we can return true only if actual timers are registered
       case ProcessingTime =>
         true
-
+      case EventTime =>
```

Comment:
We only really need to run batches if the `eviction watermark` has changed right ? something along the lines of

```
        // Process another non-data batch only if the watermark has changed in this executed plan
        eventTimeWatermarkForEviction.isDefined &&
          newInputWatermark > eventTimeWatermarkForEviction.get
```